### PR TITLE
[PRS] support choosing identity type at parallel node level.

### PR DESCRIPTION
**Pull request by bupt-wenxiaole**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by elliotzh** [comment]
@bupt-wenxiaole please add corresponding e2e tests

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 1]
```diff
@@ -179,7 +195,7 @@ def __init__(
         self.mini_batch_error_threshold = mini_batch_error_threshold
         self._resources = resources
         self.environment_variables = {} if environment_variables is None else environment_variables
-
+        self.identity = identity
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:198]: init self._identity first

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 2]
```diff
@@ -243,6 +259,45 @@ def resources(self, value):
...
+    ) -> None:
+        """Sets the identity that the job will use while running on compute.
+
+        :param value: The identity that the job will use while running on compute.
+        :type value: Union[dict[str, str], ~azure.ai.ml.ManagedIdentityConfiguration,
+            ~azure.ai.ml.AmlTokenConfiguration, ~azure.ai.ml.UserIdentityConfiguration]
+        """
+        print(f"value is {value}")
+        if isinstance(value, dict):
+            print("is dict")
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:289]: remove these logs

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 3]
```diff
@@ -243,6 +259,45 @@ def resources(self, value):
...
+            identity_schema = UnionField(
+                [
+                    NestedField(ManagedIdentitySchema, unknown=INCLUDE),
+                    NestedField(AMLTokenIdentitySchema, unknown=INCLUDE),
+                    NestedField(UserIdentitySchema, unknown=INCLUDE),
+                ]
+            )
+            value = identity_schema._deserialize(value=value, attr=None, data=None)
+            print(value)
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:298]: mark

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py [hunk 4]
```diff
@@ -243,6 +259,45 @@ def resources(self, value):
...
+            Dict[str, str], ManagedIdentityConfiguration, AmlTokenConfiguration, UserIdentityConfiguration, None
+        ],
+    ) -> None:
+        """Sets the identity that the job will use while running on compute.
+
+        :param value: The identity that the job will use while running on compute.
+        :type value: Union[dict[str, str], ~azure.ai.ml.ManagedIdentityConfiguration,
+            ~azure.ai.ml.AmlTokenConfiguration, ~azure.ai.ml.UserIdentityConfiguration]
+        """
+        print(f"value is {value}")
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py:287]: mark

**Review by shift202 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py (hunk 1 above)
Comment (bupt-wenxiaole) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel.py]: fixed

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**